test/aleph/http_test.clj

```diff
@@ -186,9 +187,20 @@
     (apply concat)
     (partition 2)))
 
+(def default-ssl-certificate
+  (SelfSignedCertificate. "localhost"))
+
+(def default-ssl-server-options
+  {:port port
+   :ssl-context (netty/ssl-server-context {:private-key       (.privateKey default-ssl-certificate)
+                                           :certificate-chain (.certificate default-ssl-certificate)})})
+
 (defmacro with-server [server & body]
   `(let [server# ~server]
-     (binding [*pool* (http/connection-pool {:connection-options (merge *connection-options* {:insecure? true})})]
+     (binding [*pool* (http/connection-pool {:connection-options (merge
+                                                                  {:ssl-context {:trust-store (.certificate default-ssl-certificate)}
+                                                                   :http-versions [:http1]}
+                                                                  *connection-options* )})]
```

Comment:
I'd suggest taking a look at aleph.http-test in the current http2 PR, because I left this alone, but built on it to make it easier to run all tests against both http versions, as needed.

This section now looks like:

```clojure
(def http1-server-options {:port port})
(def http1-ssl-server-options {:port        port
                               :ssl-context (netty/self-signed-ssl-context)})
(def http2-server-options {:port        port
                           :ssl-context ssl/http2-only-ssl-context})

(defmacro with-server [server & body]
  `(let [server# ~server]
     (binding [*pool* (http/connection-pool {:connection-options (merge *connection-options* {:insecure? true})})]
       (try
         ~@body
         (finally
           (.close ^Closeable server#)
           (netty/wait-for-close server#)
           (.shutdown *pool*))))))

(defmacro with-http1-server
  [handler server-options & body]
  `(testing "- http1"
     (with-server (http/start-server ~handler ~server-options)
       ~@body)))

(defmacro with-http2-server
  [handler server-options & body]
  ;; with-redefs used so clj fns running on netty threads will work
  `(testing "- http2"
     (with-redefs [*use-tls?* true]
       (let [server-options# ~server-options]
         (with-server (http/start-server ~handler (merge server-options# http2-server-options))
           ~@body)))))
```

Copying this, and placing your change in `with-http1-server` might cause less merge issues

Reply:
I resolved it by switching all SSL tests to use the `aleph.ssl` utils instead. Now the only diff here is the removal of `:insecure? true` which is good IMHO since it could otherwise mask SSL errors when one fails to set up the client correctly. So I think conflict-wise we should be in a better place now :pray: